ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSshell.cpp

```diff
@@ -90,20 +90,24 @@ string execute_shell_for_output(const string &command) {
     HANDLE waitHandles[] = { pi.hProcess, hStdOutPipeRead };
     while (DWORD eventSignalId = MsgWaitForMultipleObjects(2, waitHandles, false, INFINITE, QS_ALLEVENTS)) {
       if (eventSignalId == WAIT_OBJECT_0) break;
+      if (eventSignalId == WAIT_OBJECT_0 + 1) break;
       enigma::handleEvents();
       char buffer[BUFSIZ];
       DWORD dwRead = 0;
       BOOL success = ReadFile(hStdOutPipeRead, buffer, BUFSIZ, &dwRead, NULL);
       if (success || dwRead) {
         buffer[dwRead] = 0;
         output.append(buffer, dwRead);
-      }
+        printf("%s", buffer);
+      } else { break; }
     }
     CloseHandle(hStdOutPipeRead);
     CloseHandle(hStdInPipeWrite);
-    return output;
+    while (output.back() == '\r' || output.back() == '\n')
+      output.pop_back();
```

Comment:
Uh files usually end with a blank new line Sam, it's a really old standard, and without the new line file utilities can break. I don't know if it's a good idea for you to go against the norm here.
https://stackoverflow.com/questions/2287967/why-is-it-recommended-to-have-empty-line-in-the-end-of-a-source-file

Reply:
I removed that before i saw your comment, i copied code from my extension which does it, and i knew you guys didn't want it doing that because josh told me a while back. Anyways, it was an accident. It should be good now..